src/test_l3_dsVideoDevice.c

```diff
@@ -0,0 +1,687 @@
+*
+* This test provides a scope to Intialize the Video Device module.
+* **Test Group ID:** 03@n
+* **Test Case ID:** 001@n
+*
+* **Pre-Conditions:** None@n
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select VideoDevice_Init before running any test.
+*
+*/
+void test_l3_dsVideoDevice_Init()
+{
+    gTestID = 1;
+    dsError_t status   = dsERR_NONE;
+    UT_LOG_INFO("In %s [%02d%03d]", __FUNCTION__, gTestGroup, gTestID);
+
+    UT_LOG_INFO("Calling dsVideoDeviceInit()");
+    status = dsVideoDeviceInit();
+    UT_LOG_INFO("Result dsVideoDeviceInit(), dsError_t=[%s]", UT_Control_GetMapString(dsErrorMappingTable, status));
+    DS_ASSERT(status, dsERR_NONE);
+    UT_LOG_INFO("Calling dsRegisterFrameratePreChangeCB()");
+    status = dsRegisterFrameratePreChangeCB(dsVideoDevice_FrameratePreChange);
+    UT_LOG_INFO("Result dsRegisterFrameratePreChangeCB(), dsError_t=[%s]", UT_Control_GetMapString(dsErrorMappingTable, status));
+    DS_ASSERT(status, dsERR_NONE);
+    UT_LOG_INFO("Calling dsRegisterFrameratePostChangeCB()");
+    status = dsRegisterFrameratePostChangeCB(dsVideoDevice_FrameratePostChange);
+    UT_LOG_INFO("Result dsRegisterFrameratePostChangeCB(), dsError_t=[%s]", UT_Control_GetMapString(dsErrorMappingTable, status));
+    DS_ASSERT(status, dsERR_NONE);
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+* @brief Termination of the Device settings - Video Device Module
+*
+* This test provides a scope to Terminate the Video Device module.
+* **Test Group ID:** 03@n
+* **Test Case ID:** 011@n
+* **Pre-Conditions:** None@n
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select VideoDevice_Term at end of every test.
+*
+*/
+void test_l3_dsVideoDevice_Term()
+{
+    gTestID = 11;
+    UT_LOG_INFO("In %s [%02d%03d]", __FUNCTION__, gTestGroup, gTestID);
+    dsError_t status   = dsERR_NONE;
+
+    UT_LOG_INFO("Calling dsVideoDevice_Term()");
+    status = dsVideoDeviceTerm();
+    UT_LOG_INFO("Result dsVideoDeviceTerm(), dsError_t=[%s]",  UT_Control_GetMapString(dsErrorMappingTable, status));
+    DS_ASSERT(status, dsERR_NONE);
+    UT_LOG_INFO("Out %s", __FUNCTION__);
+}
+
+/**
+* @brief Set Zoom mode of Source device
+*
+* This test sets zoom mode of video device seected.
+* **Test Group ID:** 03@n
+* **Test Case ID:** 002@n
+* **Pre-Conditions:** None@n
+* VideoDevice_init Should be called before calling this test
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select SetZoomMode from list
+*
+*/
+
+void test_l3_dsVideoDevice_SetZoomMode()
+{
+    dsError_t status   = dsERR_NONE;
+    gTestID = 2;
+    UT_LOG_INFO("In %s [%02d%03d]", __FUNCTION__, gTestGroup, gTestID);
+    int32_t choice,j;
+
+    dsVideoDevice_getHandle();
+    UT_LOG_MENU_INFO(" \t  Supported Zoom Modes are:");
+    for (j = 0; j <= dsVIDEO_ZOOM_MAX; j++)
```

Comment:
initialize  j variable with the initial value of dsVIDEO_ZOOM_NONE

Reply:
done